packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -353,20 +353,22 @@ class _RawAutocompleteState<T extends Object> extends State<RawAutocomplete<T>>
   // Called when _textEditingController changes.
   Future<void> _onChangedField() async {
     final TextEditingValue value = _textEditingController.value;
+
+    // Make sure the options are no longer hidden if the content of the field
+    // changes (ignore selection changes).
+    if (value.text == _lastFieldText) {
+      return;
+    }
+    _lastFieldText = value.text;
+
     final Iterable<T> options = await widget.optionsBuilder(value);
```

Comment:
This is a bit weird: `optionsBuilder` takes a `TextEditingValue`, so any `TextEditingValue` change should send a new signal.

Reply:
Technically you can achieve the filtering by doing the filtering in the `optionsBuilder` callback I think? Although I have to admit it is not convenient.

Reply:
It wouldn't, because `optionsBuilder` will be called again before checking that `_options` have changed (so the behavior remains as is before this PR). But I agree, the onus of filtering can continue being on the user in the `optionsBuilder` callback. If we don't want the selection to cause rebuilds, then we should solve the problem once and for all by making `optionsBuilder` take only the string `textEditingController` value. So, I will go ahead and close this PR.

On a side note, https://github.com/flutter/flutter/pull/150776 does provide quite a clever solution for the debounce bug.